### PR TITLE
8301465: Remove unnecessary nmethod arming in Full GC of Serial/Parallel

### PR DESCRIPTION
**Pull request by albertnetymk**
Simple removing unnecessary code and adding API documentation.

Test: hotspot_gc

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301465](https://bugs.openjdk.org/browse/JDK-8301465): Remove unnecessary nmethod arming in Full GC of Serial/Parallel

### Reviewers
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**)
 * [Ivan Walulya](https://openjdk.org/census#iwalulya) (@walulyai - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12311/head:pull/12311` \
`$ git checkout pull/12311`

Update a local copy of the PR: \
`$ git checkout pull/12311` \
`$ git pull https://git.openjdk.org/jdk pull/12311/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12311`

View PR using the GUI difftool: \
`$ git pr show -t 12311`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12311.diff">https://git.openjdk.org/jdk/pull/12311.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/code/codeCache.cpp [hunk 1]
```diff
@@ -872,6 +872,12 @@ void CodeCache::on_gc_marking_cycle_finish() {
   update_cold_gc_count();
 }
 
+// Arm nmethods so that special actions are taken (nmethod_entry_barrier) for
+// on-stack nmethods. It's used in two places:
+// 1. Used before the start of concurrent marking so that oops inside on-stack
+//    nmethods are visited.
+// 2. Used at the end of (stw/concurrent) marking so that nmethod::_gc_epoch is
+//    up-to-date, which provides more accurate estimate of nmethod::is_cold.
```
Comment (tschatzl) [inline src/hotspot/share/code/codeCache.cpp:880]: I would prefer to have general documentation and usage information about a method in the .hpp file.

**Review by tschatzl — CHANGES_REQUESTED** [review]

File: src/hotspot/share/code/codeCache.cpp (hunk 1 above)
Comment (albertnetymk) [reply src/hotspot/share/code/codeCache.cpp]: Moved.

**Review by tschatzl — APPROVED** [review]
Lgtm.

**Review by walulyai — APPROVED** [review]

**Comment by albertnetymk** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**